update/v1alpha1/types_node_progress_insight.go

```diff
@@ -0,0 +1,161 @@
+// to hold configuration to drive what information is surfaced and how
+type NodeProgressInsightSpec struct {
+}
+
+// NodeProgressInsightStatus reports the state of a Node during the update
+// +kubebuilder:validation:XValidation:rule="self.name == self.resource.name",message=".name must match .resource.name"
+type NodeProgressInsightStatus struct {
+	// conditions provides details about the control plane update. Known conditions are:
+	// - Updating: whether the Node is updating; When Updating=False, the reason field can be Updated, Pending, or Paused. When Updating=True, the reason field can be Draining, Updating, or Rebooting
+	// - Available: whether the Node is available (accepting workloads)
+	// - Degraded: whether the Node is degraded (problem observed)
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// name is the name of the node
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// resource is the Node resource that represents the node
+	//
+	// +Note: By OpenShift API conventions, in isolation this should be a specialized reference that refers just to
+	// +resource name (because the rest is implied by status insight type). However, because we use resource references in
+	// +many places and this API is intended to be consumed by clients, not produced, consistency seems to be more valuable
+	// +than type safety for producers.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="!has(self.group) && self.resource == 'nodes'",message="resource must be a nodes.core.k8s.io resource"
+	Resource ResourceRef `json:"resource"`
+
+	// poolResource is the resource that represents the pool the node is a member of
+	//
+	// +Note: By OpenShift API conventions, in isolation this should probably be a specialized reference type that allows
+	// +only the "correct" resource types to be referenced (here, MachineConfigPool or NodePool). However, because we use
+	// +resource references in many places and this API is intended to be consumed by clients, not produced, consistency
+	// +seems to be more valuable than type safety for producers.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.group == 'machineconfiguration.openshift.io' && self.resource == 'machineconfigpools'",message="resource must be a machineconfigpools.machineconfiguration.openshift.io resource"
+	PoolResource ResourceRef `json:"poolResource"`
+
+	// scopeType describes whether the node belongs to control plane or a worker pool
+	// +required
+	Scope ScopeType `json:"scopeType"`
+
+	// version is the version of the node, when known
```

Comment:
What version is this? And, does it make sense to tie this to the Kube/OpenShift semver (looks like semver-ish match?) when Nodes are actually tied to a rendered MC version, and that can change over time even when the cluster OCP version stays the same?

Reply:
From the user PoV it makes sense - if you start updating from 4.17 to 4.18 then seeing nodes bump from 4.17 to 4.18 is what you expect. "tied to a rendered MC version" is an implementation detail that experts know.

Reply:
Ack, context in the godoc would be helpful here